src/levo_postman/app.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 Levo, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import os
+import tempfile
+
+import uvicorn as uvicorn
+from fastapi import APIRouter, FastAPI, File, UploadFile, status
+from fastapi.responses import FileResponse
+
+from .enrich_openapi_spec import enrich_spec
+from .oas_to_postman import convert_to_postman
+from .postman_collection_to_openapi import convert_to_openapi
+
+CHUNK_SIZE = 1024 * 1024  # = 1MB
+
+router = APIRouter()
+
+
+@router.get("/postman/health")
+def health_check():
+    return dict(message="I am doing good!")
+
+
+def _iter_file(file_path):
+    with open(file_path, mode="rb") as file_like:
+        yield from file_like
+
+
+@router.post("/postman/to-oas")
+async def to_oas(file: UploadFile = File(...)):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        tmp = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            tmp.write(file.file.read())
+            file_path = tmp.name
+
+            oas_file_path = await convert_to_openapi(file_path, temp_dir)
+            oas_file_path = await enrich_spec(oas_file_path)
+            return FileResponse(
+                path=oas_file_path,
+                status_code=status.HTTP_201_CREATED,
+                filename=f"{file.filename}.yaml",
+            )
+        finally:
+            tmp.close()
+            os.unlink(tmp.name)
+
+
+@router.post("/postman/from-oas")
+async def from_oas(file: UploadFile = File(...)):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        tmp = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            tmp.write(file.file.read())
+            file_path = tmp.name
+
+            postman_file_path = await convert_to_postman(file_path, temp_dir)
+            if os.stat(postman_file_path).st_size == 0:
+                raise Exception('Failed to convert OpenApi spec to Postman collection')
+
+            return FileResponse(
```

Comment:
@ricekot I'm running into a strange issue here where the `postman_file_path` does exist before we call `FileResponse` but the response processor fails to find the file. Here is the stack trace. Just curious if you can find what am I missing here?

```python
Writing to file:  true /tmp/tmp8iyryqe4/cf38d062-0222-4c75-9250-e5eb9cb10e16.json {
  result: true,
  output: [ { type: 'collection', data: [Object] } ],
  analytics: {
    actualStack: 0,
    numberOfRequests: 12,
    size: 0.004050254821777344,
    numberOfRefs: 0,
    numberOfExamples: 7,
    assignedStack: 10,
    complexityScore: 0
  }
}
Conversion successful, collection written to file
[2023-02-05 09:21:51 +0000] [7] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/venv/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 369, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/venv/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 59, in __call__
    return await self.app(scope, receive, send)
  File "/venv/lib/python3.9/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/venv/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/venv/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/venv/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/venv/lib/python3.9/site-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/venv/lib/python3.9/site-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/venv/lib/python3.9/site-packages/starlette/routing.py", line 55, in app
    await response(scope, receive, send)
  File "/venv/lib/python3.9/site-packages/starlette/responses.py", line 286, in __call__
    raise RuntimeError(f"File at path {self.path} does not exist.")
RuntimeError: File at path /tmp/tmp8iyryqe4/cf38d062-0222-4c75-9250-e5eb9cb10e16.json does not exist.
```

Reply:
This is the reason for this issue: https://stackoverflow.com/a/72535413
Fixed it now.